app/Http/Controllers/Playlist/PlaylistsController.php

```diff
@@ -699,6 +679,53 @@ private function processVersesOnTransportStream($item, $transportStream, $bible_
         return $transportStream;
     }
 
+    private function getHlsPlaylist($response, $items, $download)
+    {
+        $signed_files = [];
+        $transaction_id = random_int(0, 10000000);
+        try {
+            apiLogs(request(), $response->getStatusCode(), $transaction_id);
+        } catch (\Exception $e) {
+            Log::error($e);
+        }
+        $durations = [];
+        $hls_items = '';
+        foreach ($items as $item) {
+            $fileset = $item->fileset;
+            if (!Str::contains($fileset->set_type_code, 'audio')) {
+                continue;
+            }
+            $bible_files = BibleFile::with('streamBandwidth.transportStreamTS')->with('streamBandwidth.transportStreamBytes')->where([
+                'hash_id' => $fileset->hash_id,
+                'book_id' => $item->book_id,
+            ])
+                ->where('chapter_start', '>=', $item->chapter_start)
+                ->where('chapter_start', '<=', $item->chapter_end)
+                ->get();
+            if ($fileset->set_type_code === 'audio_stream' || $fileset->set_type_code === 'audio_drama_stream') {
+                $result = $this->processHLSAudio($bible_files, $hls_items, $signed_files, $transaction_id, $item, $download);
+                $hls_items = $result->hls_items;
```

Comment:
here and on line 712 it seems like hls_items should be appended (.=), not replaced (=, which seems like it would result in playlists having only one item)?

Reply:
and similarly for signed_files

Reply:
The function `processHLSAudio` and `processMp3Audio` returns `$result->hls_items` and `$result->signed_files` with the new content added to the previous one. For this PR I didn't change the code of this function. 

```php
private function processHLSAudio($bible_files, $hls_items, $signed_files, $transaction_id, $item, $download)
    {
        $durations = [];
        foreach ($bible_files as $bible_file) {
            $currentBandwidth = $bible_file->streamBandwidth->first();

            $transportStream = sizeof($currentBandwidth->transportStreamBytes) ? $currentBandwidth->transportStreamBytes : $currentBandwidth->transportStreamTS;
            if ($item->verse_end && $item->verse_start) {
                $transportStream = $this->processVersesOnTransportStream($item, $transportStream, $bible_file);
            }

            $fileset = $bible_file->fileset;

            foreach ($transportStream as $stream) {
                $durations[] = $stream->runtime;
                $hls_items .= "\n#EXTINF:$stream->runtime," . $item->id;
                if (isset($stream->timestamp)) {
                    $hls_items .= "\n#EXT-X-BYTERANGE:$stream->bytes@$stream->offset";
                    $fileset = $stream->timestamp->bibleFile->fileset;
                    $stream->file_name = $stream->timestamp->bibleFile->file_name;
                }
                $bible_path = $bible_file->fileset->bible->first()->id;
                $file_path = 'audio/' . $bible_path . '/' . $fileset->id . '/' . $stream->file_name;
                if (!isset($signed_files[$file_path])) {
                    $signed_files[$file_path] = $this->signedUrl($file_path, $fileset->asset_id, $transaction_id);
                }
                $hls_file_path = $download ? $file_path : $signed_files[$file_path];
                $hls_items .= "\n" . $hls_file_path;
            }
            $hls_items .= "\n" . '#EXT-X-DISCONTINUITY';
        }

        return (object) ['hls_items' => $hls_items, 'signed_files' => $signed_files, 'durations' => $durations];
    }

    private function processMp3Audio($bible_files, $hls_items, $signed_files, $transaction_id, $download, $item)
    {
        $durations = [];
        foreach ($bible_files as $bible_file) {
            $default_duration = $bible_file->duration ?? 180;
            $durations[] = $default_duration;
            $hls_items .= "\n#EXTINF:$default_duration," . $item->id;

            $bible_path = $bible_file->fileset->bible->first()->id;
            $file_path = 'audio/' . $bible_path . '/' . $bible_file->fileset->id . '/' . $bible_file->file_name;
            $hls_items .= "\n";
            if (!isset($signed_files[$file_path])) {
                $signed_files[$file_path] = $this->signedUrl($file_path, $bible_file->fileset->asset_id, $transaction_id);
            }
            $hls_file_path = $download ? $file_path : $signed_files[$file_path];
            $hls_items .= "\n" . $hls_file_path;
            $hls_items .= "\n" . '#EXT-X-DISCONTINUITY';
        }

        return (object) ['hls_items' => $hls_items, 'signed_files' => $signed_files, 'durations' => $durations];
    }
```

Reply:
oh i see, thx